SMF records feature in API ML (#2536)
zowe/docs-site: docs/user-guide/api-mediation/api-mediation-smf.md

# SMF records

> API Mediation Layer can issue SMF type 83, 230, or 231 security-related audit records. This requires the generation of a Personal Access Token.

Reviewer: I think it would be useful to specify the use case for issuing these types of SMF records.
Something like, "You can use SMF records to assist with analyzing configuration scheduling jobs, evaluating data set activity, profiling system resource use and maintaining system security."

Reviewer 2: I agree. `This requires the generation of a Personal Access Token.` is the use case. I would not say it is requirement. It is rather the first use case.

Otherwise, I can approve it.

Author: Well, mine original sentence was this one, without any "requires", that @janan07 changed:
https://github.com/zowe/docs-site/blob/0647cd355e42b16bb2516a41696c816e45baeace/docs/user-guide/api-mediation/api-mediation-smf.md?plain=1#L3

Author: I updated sentence. Please, take a look.

# SMF records

> API Mediation Layer can issue an SMF type 83 sybtype 2 security related audit records only when Personal Access Token is generated.

Reviewer 3: ```suggestion
API Mediation Layer can issue an SMF type 83, 230, or 231 security-related audit records only when Personal Access Token is generated.
```

The subtype is configurable and 83 is not accurate.

> To enable this functionality on your Zowe instance follow this configuration guide:

Reviewer 3: ```suggestion
To enable this functionality on your Zowe instance follows this configuration guide:
```

[Configure main Zowe server to issue SMF records](#configure-main-zowe-server-to-issue-smf-records)
   * [Using RACF](#using-racf)
   * [Using ACF2](#using-acf2)
   * [Using TSS](#using-tss)

> Some of predefined values in SMF record is possible to change. See, the [full list of configurable parameters](#smf-record-configurable-parameters). 

Reviewer 3: ```suggestion
Some of the predefined values in the SMF record are possible to change. See, the [full list of configurable parameters](#smf-record-configurable-parameters). 
```

> **Note:** Record type 83 is a RACF processing record, and it can be wrapped into different SMF type depends on ESM:

Reviewer 3: ```suggestion
**Note:** Record type 83 is a RACF processing record, and it can be replaced by a different SMF type depending on ESM:
```

* ACF2 - SMF type 230
* TSS - SMF type 231

> ## Configure main Zowe server to issue SMF records

Reviewer 3: ```suggestion
## Configure the main Zowe server to issue SMF records
```

This security configuration is necessary for API ML to be able to issue SMF records. A user running API Gateway must have read access to the RACF general resource `IRR.RAUDITX` in the `FACILITY` class.
To set up this security configuration, submit the `ZWESECUR` JCL member. For users upgrading from version 1.18 and lower use the following configuration steps.

### Using RACF

If you use RACF, verify and update permission in the `FACILITY` class.

**Follow these steps:**

1. Verify user `ZWESVUSR` has read access.

    ```
    RLIST FACILITY IRR.RAUDITX AUTHUSER
    ```

> 

Reviewer 2: ```suggestion
Review the output to confirm that the profile exists and that the user ZWESVUSR who runs the ZWESVSTC started task has READ access to this profile.

```

2. Add user `ZWESVUSR` permission to read.
    ```
    PERMIT IRR.RAUDITX CLASS(FACILITY) ACCESS(READ) ID(ZWESVUSR)
    ```
3. Activate changes.
    ```
    SETROPTS RACLIST(FACILITY) REFRESH
    ```

### Using ACF2

> If you use ACF2, verify and update permission in the `FACILITY` class.

Reviewer 2: Is it really update?

**Follow these steps:**

1. Verify user `ZWESVUSR` has read access.
    ```      
    SET RESOURCE(FAC) 
    LIST LIKE(IRR-)
    ```    
2. Add user `ZWESVUSR` permission to read.
    ```
    RECKEY IRR.RAUDITX ADD(SERVICE(READ) ROLE(&STCGRP.) ALLOW)
    ```

### Using TSS

If you use TSS, verify and update permission in `FACILITY` class.

**Follow these steps:**

1. verify user `ZWESVUSR` has read access.
    ```      
    TSS WHOHAS IBMFAC(IRR.RAUDITX)
    ```    
2. Add user `ZWESVUSR` permission to read.
    ```
    TSS PER(ZWESVUSR) IBMFAC(IRR.RAUDITX) ACCESS(READ)
    ```
   
## SMF record configurable parameters

> The following list of parameters is required to set up SMF records, and the default values of these parameters can be overwritten in `zowe.yaml`. See [how to configure rauditx parameters](#configure-rauditx-parameters).

Reviewer 2: ```suggestion
The following list of parameters can be used to modify the default SMF record values. The default values of these parameters can be overwritten in `zowe.yaml`. See [how to configure rauditx parameters](#configure-rauditx-parameters).
```

| rauditx.subtype           | SMF type 83 record subtype assigned to the component. For more information, see [description of subtypes](https://www.ibm.com/docs/en/zos/2.5.0?topic=records-record-type-83-security-events) | integer |       2       |
| rauditx.event             | Event code. For more information, see [description of event codes](https://www.ibm.com/docs/en/zos/2.5.0?topic=descriptions-event-codes-event-code-qualifiers)                                | integer |       2       |
| rauditx.qualifier.success | Event Code Qualifier for success. The value can be between 0 and 255                                                                                                                          | integer |       0       |
| rauditx.qualifier.failed  | Event Code Qualifier for failure. The value can be between 0 and 255                                                                                                                          | integer |       1       |

### Configure rauditx parameters

> Example of changing `rauditx.fmid` parameter. Following procedure can be applied for any of SMF record configurable parameters.

Reviewer 3: ```suggestion
Example of changing `rauditx.fmid` parameter. The following procedure can be applied to any SMF record configurable parameters.
```

**Follow these steps:**

1. Open the `zowe.yaml` configuration file.

> 2. Find or add the property `zowe.environments.RAUDITX_FMID` and set your desired value.

Reviewer 3: Are we sure about the mapping of this value to the used property in the bean?

Author: Well, it was working for `apiml.security.saf.provider` untill in api-layer they added this property to `start.sh`.
